### PR TITLE
better docs: options

### PR DESCRIPTION
**Pull request by narimiran**

### CODE REVIEW

File: lib/pure/options.nim [hunk 1]
```diff
@@ -41,16 +38,12 @@
...
-## from ``system.Defect``, and should therefore never be catched. Instead, rely on
-## checking if the option contains a value with ``isSome`` and ``isNone``.
 ##
-## There is another option for obtaining the value: ``unsafeGet``, but you must
-## only use it when you are absolutely sure the value is present (e.g. after
-## checking ``isSome``). If you do not care about the tiny overhead that ``get``
-## causes, you should simply never use ``unsafeGet``.
+## The ``get`` operation demonstrated above returns the underlying value, or
+## raises ``UnpackError`` if there is no value. Note that ``UnpackError``
+## inherits from ``system.Defect``, and should therefore never be catched.
```
Comment (citycide) [inline lib/pure/options.nim]: `catched` is very strange, maybe `caught`?

```suggestion
## inherits from ``system.Defect``, and should therefore never be caught.
```

File: lib/pure/options.nim [hunk 2]
```diff
@@ -79,123 +72,302 @@ type
...
+    doAssertRaises(UnpackError):
+      echo b.get
+
   if self.isNone:
     raise newException(UnpackError, "Can't obtain a value from a `none`")
   self.val
 
 proc get*[T](self: Option[T], otherwise: T): T =
-  ## Returns the contents of this option or `otherwise` if the option is none.
+  ## Returns the contents of the ``Option`` or an `otherwise` value if
```
Comment (citycide) [inline lib/pure/options.nim]: While single-backtick code markup is supported now double vs single should be consistent in Nim's docs IMO.

Single has the benefit of being easier to turn into an RST style link, aesthetics/ergonomics aside.

File: lib/pure/options.nim (hunk 1 above)
Comment (narimiran) [reply lib/pure/options.nim]: It was there from before :) Will change it.

File: lib/pure/options.nim (hunk 2 above)
Comment (narimiran) [reply lib/pure/options.nim]: The existing docs was a mixture of single- and double-backticks, so I decided to keep it that way.

But yeah, I'll change everything to single-backticks to be consistent.

**Pull request merged**